src/Illuminate/Cache/DatabaseStore.php

```diff
@@ -149,23 +145,32 @@ public function decrement($key, $value = 1)
      * @param  string  $key
      * @param  mixed  $value
      * @param  \Closure  $callback
-     * @return void
+     * @return int|bool
      */
     protected function incrementOrDecrement($key, $value, Closure $callback)
     {
-        $prefixed = $this->prefix.$key;
+        return $this->connection->transaction(function () use ($key, $value, $callback) {
+            $prefixed = $this->prefix.$key;
 
-        $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+            $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+
```

Comment:
process 1 reads value with shared lock
process 2 reads value with shared lock
process 1 updates value
process 2 already ready the old value, increments it back to the previous
value.

On Wed, Dec 30, 2015 at 9:29 AM, Will Rowe <notifications@github.com> wrote:

> In src/Illuminate/Cache/DatabaseStore.php
> <https://github.com/laravel/framework/pull/11582#discussion_r48609726>:
>
> >
> > -        $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
> > +            $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
> > +
>
> Do you mean if increment or decrement are called simultaneously, or are
> you thinking of another case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/11582/files#r48609726>.
>